core/src/main/java/hudson/Util.java

```diff
@@ -252,17 +258,11 @@ public static void deleteFile(@Nonnull File f) throws IOException {
             if(!f.delete() && f.exists()) {
                 // trouble-shooting.
                 try {
-                    Class.forName("java.nio.file.Files").getMethod("delete", Class.forName("java.nio.file.Path")).invoke(null, File.class.getMethod("toPath").invoke(f));
-                } catch (InvocationTargetException x) {
-                    Throwable x2 = x.getCause();
-                    if (x2 instanceof IOException) {
-                        // may have a specific exception message
-                        throw (IOException) x2;
-                    }
-                    // else suppress
-                } catch (Throwable x) {
-                    // linkage errors, etc.; suppress
+                    Files.deleteIfExists(f.toPath());
+                } catch(Exception e) {
+                    // caught to help further analysis, see below
```

Comment:
re-reading this I think you where correct the first time, this code would be hit for the JDK6 case but never the JDK7 case which would throw the underlying IOException which should have [details](https://github.com/jenkinsci/jenkins/pull/2256/commits/4d900fff7a03c4b026239306deadff67ae2e9e5e#diff-6b2c22336dd63b20019b8558c7e9a13fL259)  Sorry ;(